project/Bootstrap.scala

```diff
@@ -0,0 +1,39 @@
+package sbt
+
+import sbt.internal.classpath.ClassLoaderCache
+import sbt.internal.inc.ScalaInstance
+
+object Bootstrap {
+  // copied from https://github.com/sbt/sbt/blob/33e2bfe2810246c8e4e1b37ec110d15e588a4fce/main/src/main/scala/sbt/Defaults.scala#L1120-L1165
```

Comment:
Could this method be made public upstream?

Reply:
Don't think that's worth it. It's only useful for works on the Scala compilers and we probably don't want to maintain compatibility just in case other people use it.

Reply:
Your call, but It seems better than having to deal with the complicated classloader caching logic API directly. And does this need to be in `package sbt`?

Reply:
It does because the `ScalaInstance` constructor is package private. But we could as well create our own `ScalaInstance` implementation.

This code is a quick and dirty copy-paste, it can be made simpler.

Reply:
> It does because the ScalaInstance constructor is package private. 

It looks public to me (https://github.com/sbt/zinc/blob/13ddb10938237263f706174439d8ed055e8e4ac2/internal/zinc-classpath/src/main/scala/sbt/internal/inc/ScalaInstance.scala#L34-L43), and weren't we already using it before this PR?